src/features/SendAssets/components/SendAssetsForm.tsx

```diff
@@ -361,432 +363,383 @@ const SendAssetsForm = ({ txConfig, onComplete }: IStepComponentProps) => {
     []
   );
 
-  return (
-    <div>
-      <Formik
-        initialValues={initialValues}
-        validationSchema={SendAssetsSchema}
-        onSubmit={(fields) => {
-          onComplete(fields);
-        }}
-        children={({ errors, setFieldValue, setFieldTouched, touched, values }) => {
-          useEffect(() => {
-            if (updateFormValues) {
-              updateFormValues(values);
-            }
-          }, [values]);
+  const formik = useFormik({
+    initialValues,
+    validationSchema: SendAssetsSchema,
+    onSubmit: (fields) => {
+      onComplete(fields);
+    }
+  });
+
+  const { values, setFieldValue, setFieldTouched, setFieldError, errors, touched } = formik;
+
+  useEffect(() => {
+    if (updateFormValues) {
+      updateFormValues(values);
+    }
+  }, [values]);
+
+  useEffect(() => {
+    handleNonceEstimate(values.account);
+  }, [values.account]);
+
+  // Set gas estimates if default asset is selected
+  useEffectOnce(() => {
+    if (!isEmpty(values.asset)) {
+      fetchGasPriceEstimates(values.network).then((data) => {
+        setFieldValue('gasEstimates', data);
+        setFieldValue('gasPriceSlider', data.fast);
+      });
+    }
+  });
+
+  useEffect(() => {
+    if (ptxState.protectTxShow) {
+      if (goToInitialStepOrFetchReport && ptxState.receiverAddress !== values.address.value) {
+        const { address, network } = values;
+        goToInitialStepOrFetchReport(address.value, network);
+      }
+    }
+  }, [values.address.value]);
+
+  const toggleAdvancedOptions = () => {
+    setFieldValue('advancedTransaction', !values.advancedTransaction);
+  };
+  const toggleIsAutoGasSet = () => {
+    // save value because setFieldValue method is async and values are not yet updated	            // save value because setFieldValue method is async and values are not yet updated
+    const isEnablingAutoGas = !values.isAutoGasSet;
+    setFieldValue('isAutoGasSet', !values.isAutoGasSet);
+    if (isEnablingAutoGas) {
+      handleGasEstimate(true);
```

Comment:
Move to effect

Reply:
This seems very specific as it uses the `forceEstimate`, not sure whether its a good idea to move to a shared effect.

Reply:
We could use a separate state flag to determine if we need to force estimate.
The guiding principle is how difficult will it be to test that gas is correctly recalculated of toggle?